packages/relic/lib/src/headers/typed/headers/accept_language_header.dart

```diff
@@ -0,0 +1,86 @@
+import 'package:relic/src/headers/extension/string_list_extensions.dart';
+import 'package:relic/src/headers/typed/base/typed_header.dart';
+
+/// A class representing the HTTP Accept-Language header.
+///
+/// This header specifies the natural languages that are preferred in the response.
+class AcceptLanguageHeader extends TypedHeader {
```

Comment:
Just curious:
 * Have you considered [extension types](https://dart.dev/language/extension-types) instead of creating an object for each header?
   * I get the impression that you'll be parsing headers that aren't even read.
   * It might be cheaper, then again might be slightly more dynamic.

Nit: Have you considered using [class modifiers](https://dart.dev/language/class-modifiers) to prevent users from implementing these class, thus, ensuring that you can extend the interface in the future, should you ever want to.

Reply:
Hi @jonasfj,  
This update aims to eliminate dynamic typing entirely, which is the primary reason for this PR.

Reply:
> This is something we are aiming to fix in upcoming changes; if strict mode is disabled, we should only parse headers that are actually accessed.

I suspect that if you do this, you might need to change the contract around what a `Handler` is.

I'm guessing that like in `shelf` a `Handler` is a function that takes a `Request` and returns a `FutureOr<Response>`.

But if you're parsing headers on-the-fly, you'll have a harder time communicating a 400 response from inside the request handler.

When working on `pub.dev` we found that having a `ResponseException` was pretty handy.

Basically, we made some middelware that would catch `ResponseException` and return `ResponseException.response`.

Then whenever we try to parse a header on-the-fly and it fails, we just throw a `ResponseException(400)` or if authorization failed throw `ResponseException(401)`.

This contract isn't part of `shelf`, and maybe it shouldn't be, but if you want to do on-the-fly parsing of headers and reject requests with invalid headers, you might need to make such a concept part of the `Handler` specification in relic.

(just some food for thought).